src/o365/spo/commands/folder/folder-move.ts

```diff
@@ -0,0 +1,340 @@
+   */
+  progressMaxPollAttempts: number;
+  /**
+   * Retry attempts before give up.
+   * Give up if /_api/site/GetCopyJobProgress returns 
+   * X reject promises in a row
+   */
+  progressRetryAttempts: number;
+}
+
+class SpoFolderMoveCommand extends SpoCommand {
+  public get name(): string {
+    return commands.FOLDER_MOVE;
+  }
+
+  public get description(): string {
+    return 'Moves a folder to another location';
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    const resource: string = Auth.getResourceFromUrl(args.options.webUrl);
+    let siteAccessToken: string = '';
+    const webUrl: string = args.options.webUrl;
+    const tenantUrl: string = `${url.parse(webUrl).protocol}//${url.parse(webUrl).hostname}`;
+
+    if (this.debug) {
+      cmd.log(`Retrieving access token for ${resource}...`);
+    }
+
+    auth
+      .getAccessToken(resource, auth.service.refreshToken as string, cmd, this.debug)
+      .then((accessToken: string): request.RequestPromise => {
+        if (this.debug) {
+          cmd.log(`Retrieved access token ${accessToken}.`);
+        }
+
+        siteAccessToken = accessToken;
+
+        const sourceAbsoluteUrl: string = this.urlCombine(webUrl, args.options.sourceUrl);
+        const ignoreVersionHistory: boolean = args.options.ignoreVersionHistory || true;
+        const allowSchemaMismatch: boolean = args.options.allowSchemaMismatch || false;
+        const requestUrl: string = `${webUrl}/_api/site/CreateCopyJobs`;
+        const requestOptions: any = {
+          url: requestUrl,
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${siteAccessToken}`,
+            'accept': 'application/json;odata=nometadata'
+          }),
+          body: {
+            exportObjectUris: [sourceAbsoluteUrl],
+            destinationUri: this.urlCombine(tenantUrl, args.options.targetUrl),
+            options: { 
+              "IgnoreVersionHistory": ignoreVersionHistory,
+              "AllowSchemaMismatch": allowSchemaMismatch,
+              "IsMoveMode":true, 
+            }
+          },
+          json: true
+        };
+
+        if (this.debug) {
+          cmd.log('CreateCopyJobs request...');
+          cmd.log(requestOptions);
+        }
+
+        return request.post(requestOptions);
+      })
+      .then((jobInfo: any): Promise<any> => {
+        if (this.debug) {
+          cmd.log('CreateCopyJobs response...');
+          cmd.log(jobInfo);
+        }
+
+        const jobProgressOptions: JobProgressOptions = {
+          webUrl: webUrl,
+          accessToken: siteAccessToken,
+          copyJopInfo: jobInfo.value[0],
+          progressMaxPollAttempts: 1000, // 1 sec.
+          progressPollInterval: 30 * 60, // approx. 30 mins. if interval is 1000
+          progressRetryAttempts: 5
+        }
+
+        return this.getCopyJobProgress(jobProgressOptions, cmd);
+      })
+      .then((resp: any): void => {
```

Comment:
If `getCopyJobProgress` returns `Promise<void>`, then `resp` is not `any` but undefined, so we should use an empty function instead.

Reply:
```suggestion
      .then((): void => {
```